libs/deal-ticket/src/components/fees-breakdown/fees-breakdown.tsx

```diff
@@ -43,85 +43,95 @@ const FeesBreakdownItem = ({
 );
 
 export const FeesBreakdown = ({
+  totalFeeAmount,
   fees,
   feeFactors,
   symbol,
   decimals,
+  referralDiscountFactor,
+  volumeDiscountFactor,
 }: {
+  totalFeeAmount?: string;
   fees?: TradeFee;
   feeFactors?: FeeFactors;
   symbol?: string;
   decimals: number;
+  referralDiscountFactor?: string;
+  volumeDiscountFactor?: string;
 }) => {
-  if (!fees) return null;
-  const totalFees = sumFees(fees);
-  const {
-    total: totalDiscount,
-    referral: referralDiscount,
-    volume: volumeDiscount,
-  } = sumFeesDiscounts(fees);
-  if (totalFees === '0') return null;
+  if (!fees || !totalFeeAmount || totalFeeAmount === '0') return null;
+
+  const { discountedFee: discountedInfrastructureFee } = getDiscountedFee(
+    fees.infrastructureFee,
+    referralDiscountFactor,
+    volumeDiscountFactor
+  );
+
+  const { discountedFee: discountedLiquidityFee } = getDiscountedFee(
```

Comment:
Might as well rename these as useDiscountedFee so we get a better experience in react dev tools

Reply:
Nevermind on this one. Bigger changes would be required due to conditional rendering of hooks